openmdao/core/system.py

```diff
@@ -2038,9 +2038,27 @@ def resolve(to_match, io_types, matches, proms):
                     call = 'promotes'
                 else:
                     call = 'promotes_%ss' % io_types[0]
-                raise RuntimeError("%s: '%s' failed to find any matches for the following "
-                                   "names or patterns: %s.%s" %
-                                   (self.msginfo, call, sorted(not_found), empty_group_msg))
+
+                for p in patterns:
+                    for name, alias in renames.items():
+                        if fnmatchcase(name, p):
+                            raise RuntimeError("%s: %s '%s' matched '%s'. You are attempting to "
```

Comment:
Since we have the information available here, we should maybe say " ... matched 'bb' but 'bb' was aliased to 'xx'."

Reply:
fixed